include/xtensor/xpad.hpp

```diff
@@ -0,0 +1,68 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay,  Wolf Vollprecht and  *
+* Martin Renou                                                             *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_PAD_HPP
+#define XTENSOR_PAD_HPP
+
+#include "xarray.hpp"
+#include "xtensor.hpp"
+#include "xview.hpp"
+#include "xstrided_view.hpp"
+
+namespace xt
+{
+    /**
+     * @brief Defines different algorithms to be used in ``xt::pad``
+     */
+    enum class pad_mode
+    {
+        periodic
+    };
+
+    template <class E, class S>
+    auto pad(E&& e, std::initializer_list<S> pad_width, pad_mode mode = pad_mode::periodic)
+    {
+        auto out = e;
```

Comment:
``auto`` drops the reference, so here you make a copy of `e`. Later, you assign a generator holding a reference on `out` to `out` itself, which is not optimal and has aliases problems.